src/libraries/System.Globalization/tests/Invariant/InvariantMode.cs

```diff
@@ -1206,6 +1206,41 @@ public void TestCultureInfo_Ctor_Int32_ReturnsInvariant(int culture)
             Assert.Equal(new CultureInfo(culture), CultureInfo.InvariantCulture);
         }
 
+        [Fact]
+        public void TestCasing()
+        {
+            // This test verifies that the casing table in the invariant globalization mode.
+            // The casing table is what we generate from the Unicode data files, and implemented in the CharUnicodeInfo.cs file.
+
+            TextInfo textInfo = CultureInfo.InvariantCulture.TextInfo;
+
+            for (int i = 0; i <= 0xFFFF; i++)
+            {
+                char ch = (char)i;
+                char upper = textInfo.ToUpper(ch);
+                char lower = textInfo.ToLower(ch);
+
+                if (ch != upper)
+                {
+                    string upperString = upper.ToString();
+                    string chString = ch.ToString();
+
+                    Assert.True(chString.Equals(upperString, StringComparison.OrdinalIgnoreCase), $"Expected {(int)ch:x4} to be equal to {(int)upper:x4}.");
+                    Assert.True(chString.IndexOf(upperString, StringComparison.OrdinalIgnoreCase) == 0, $"Expected {(int)ch:x4} exist in {(int)upper:x4}.");
```

Comment:
Since `StartsWith` is also mentioned in the issue #89591, should we `Assert.True(chString.StartsWith(upperString, StringComparison.OrdinalIgnoreCase), $"Expected {(int)ch:x4} start with {(int)upper:x4}.");` them as well?

*note, we probably cannot safely concatenate something to the `chString because there could be some RTL triggering going on... maybe?